src/pages/workspaces/workspace/analysis/Analyses.test.ts

```diff
@@ -0,0 +1,386 @@
+  appDataDisks: [],
+  persistentDisks: []
+}
+
+const defaultUseAnalysisStore = {
+  refreshFileStore: () => Promise.resolve(),
+  loadedState: { status: 'Ready', state: [] as AnalysisFile[] } as LoadedState<AnalysisFile[], unknown>,
+  create: () => Promise.resolve(),
+  pendingCreate: { status: 'None', state: true } as LoadedState<true, unknown>,
+  pendingDelete: { status: 'None', state: true } as LoadedState<true, unknown>,
+  deleteFile: () => Promise.resolve()
+}
+
+const defaultAnalysesProps: AnalysesProps = {
+  workspace: defaultGoogleWorkspace,
+  analysesData: defaultAnalysesData,
+  onRequesterPaysError: () => {},
+  storageDetails: { googleBucketLocation: '', googleBucketType: '' }
+}
+
+
+const watchCaptureEvent = jest.fn()
+type AjaxContract = ReturnType<typeof Ajax>;
+type AjaxMetricsContract = AjaxContract['Metrics']
+type OuterWorkspacesContract = AjaxContract['Workspaces']
+type InnerWorkspacesContract = AjaxContract['Workspaces']['workspace']
+const mockInnerWorkspaces = jest.fn().mockReturnValue({
+  listActiveFileTransfers: jest.fn()
+}) as InnerWorkspacesContract
+const mockOuterWorkspaces: Partial<OuterWorkspacesContract> = {
+  workspace: mockInnerWorkspaces
+}
+
+const mockMetrics: Partial<AjaxMetricsContract> = {
+  captureEvent: (event, details) => watchCaptureEvent(event, details)
+}
+
+const defaultAjaxImpl: Partial<AjaxContract> = {
+  Workspaces: mockOuterWorkspaces as OuterWorkspacesContract,
+  Metrics: mockMetrics as AjaxMetricsContract
+}
+
+describe('Analyses', () => {
+  beforeEach(() => {
+    // Arrange
+    asMockedFn(useAnalysisFiles).mockImplementation(() => defaultUseAnalysisStore)
+    asMockedFn(notebookLockHash).mockImplementation(() => Promise.resolve('testhash'))
+    asMockedFn(findPotentialNotebookLockers).mockImplementation(() => Promise.resolve({}))
+    asMockedFn(Ajax).mockImplementation(() => defaultAjaxImpl as AjaxContract)
+    asMockedFn(getLocalPref).mockImplementation(() => undefined)
+  })
+
+  it('loads properly with no files', async () => {
+    // Act
+    await act(async () => {
+      await render(h(BaseAnalyses, defaultAnalysesProps))
+    })
+
+    // Assert
+    screen.getByText('Start')
+    // Banner text and images
+    screen.getByText('A place for all your analyses')
+    screen.getByAltText('Jupyter')
+    screen.getByAltText('RStudio Bioconductor')
+    screen.getByAltText('Galaxy')
+  })
+
+  it('loads properly with files for a google workspace', async () => {
+    // Arrange
+    const fileName1 = 'file1.ipynb'
+    const fileName2 = 'file2.ipynb'
+    const files = [getFileFromPath(`test/${fileName1}` as AbsolutePath), getFileFromPath(`test/${fileName2}` as AbsolutePath)]
+
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      ...defaultUseAnalysisStore,
+      loadedState: { status: 'Ready', state: files }
+    }))
+
+    // Act
+    await act(async () => {
+      await render(h(BaseAnalyses, defaultAnalysesProps))
```

Comment:
Hmm, so here the base component itself does not have a well-formed on-mount that performs state updates. The correct way to do this would be to move the onMount side effects to a hook, like we've done for past components with this issue. This seems like a bit much for the scope of this PR, so for now awaiting the render waits long enough for the re-render once onMount updates the state. Would be interested to pair if you have a better way to do this!

Reply:
Because of the async state updates in onMount, the `act` function has to be async and awaited. But `render` itself does not need to be awaited (because it does not return a promise).
```js
await act(async () => {
  render(h(BaseAnalyses, defaultAnalysesProps))
})
```

ESLint complains about this, but that can be silenced with a comment.